src/VisualStudio/CSharp/Impl/ProjectSystemShim/CSharpProjectShimWithServices.CodeModel.cs

```diff
@@ -36,36 +36,18 @@ protected override void OnDocumentRemoved(string filePath)
             base.OnDocumentRemoved(filePath);
 
             // We may have a code model floating around for it
-            var codeModelCache = ProjectCodeModel.GetCodeModelCache();
-            if (codeModelCache != null)
-            {
-                codeModelCache.OnSourceFileRemoved(filePath);
-            }
+            ProjectCodeModel.OnSourceFileRemoved(filePath);
         }
 
         public override int CreateCodeModel(object parent, out EnvDTE.CodeModel codeModel)
         {
-            var codeModelCache = ProjectCodeModel.GetCodeModelCache();
-            if (codeModelCache == null)
-            {
-                codeModel = null;
-                return VSConstants.E_FAIL;
```

Comment:
Do remember that there was significant effort made to return the same HRESULTs to avoiding breaking downstream components that depend on those failure codes. This might be overkill here, but it's something to keep in mind.

Reply:
For the CodeModelCache == null cases, I don't believe it's possible to ever hit that. I'm sure by writing that I'll now find myself proved wrong, but this would be less risky than say all those crazy returns for GetStartPoint/GetEndPoint.